lib/wasi.js

```diff
@@ -109,11 +109,13 @@ class WASI {
       wrap[prop] = FunctionPrototypeBind(wrap[prop], wrap);
     }
 
+    let returnOnExit = true;
     if (options.returnOnExit !== undefined) {
-      validateBoolean(options.returnOnExit, 'options.returnOnExit');
-      if (options.returnOnExit)
-        wrap.proc_exit = FunctionPrototypeBind(wasiReturnOnProcExit, this);
+      returnOnExit = options.returnOnExit;
+      validateBoolean(returnOnExit, 'options.returnOnExit');
     }
+    if (returnOnExit === true)
```

Comment:
```suggestion
    if (returnOnExit)
```
Since we're already validating that it's a boolean.